source/common/grpc/typed_async_client.h

```diff
@@ -0,0 +1,114 @@
+#pragma once
+
+#include <chrono>
+
+#include "envoy/grpc/async_client.h"
+
+namespace Envoy {
+namespace Grpc {
+
+/**
+ * Forward declarations for helper functions.
+ */
+void SendMessageUntyped(RawAsyncStream* stream, const Protobuf::Message& request, bool end_stream);
+ProtobufTypes::MessagePtr ParseMessageUntyped(ProtobufTypes::MessagePtr&& message,
+                                              Buffer::InstancePtr&& response);
+RawAsyncStream* StartUntyped(RawAsyncClient* client,
```

Comment:
Nit: please lowerCase all method names in this PR.

Reply:
done. AFAICT